cvae/README.md

```diff
@@ -0,0 +1,157 @@
+# Convolutional Variational Autoencoder (CVAE) on MNIST
+
+## About
+
+Convolutional variational autoencoder (CVAE) implementation in MLX using MNIST. This is a minimal example ported from the [torch-vae](https://github.com/menzHSE/torch-vae) repo. 
+
+
+## Variational Autoencoder Implementation Overview
+
+Good overviews of variational autoencoders are provided in [arXiv:1906.02691](https://arxiv.org/abs/1906.02691) and [arXiv:1312.6114](https://arxiv.org/abs/1312.6114).
+
+In our implementation, the input image is not directly mapped to a single latent vector. Instead, it is transformed into a probability distribution within the latent space, from which we sample a latent vector for reconstruction. The process involves:
```

Comment:
Could you say a bit more about this? Isn't this what a VAE always does? Or did you do something non-standard there?

Reply:
That's a standard approach called "reparameterization trick", see https://arxiv.org/pdf/1312.6114.pdf (Sec. 2.4). It essentially makes the sampling process differentiable.